File: Project Euler #9/CPP/solution9.cpp
#include <iostream>

using namespace std;


int main() {
    /* Enter your code here. Read input from STDIN. Print output to STDOUT */ 
    int t;
    cin>>t;
    while(t--){
        int n;
        cin>>n;
        int large = 0;
        for(int a=3;a<=n/3;a++){
            int b = (n*n-2*n*a)/(2*n-2*a);
            int c = n-a-b;
            if(a*a+b*b == c*c && a+b+c==n){
                if(a*b*c>large)
                    large = a*b*c;
            }
        }
        if(large != 0)
            cout<<large<<endl;
        else
            cout<<-1<<endl;    
    }  
    return 0;
}

File: Project Euler #9/Python/solution9.py

if __name__ == "__main__":
	t = int(input().strip())
	for a0 in range(t):
	    n = int(input().strip())
	    large = 0
	    for a in range(1,n//3,1):
	        b = (n*n-2*n*a)//(2*n-2*a)
	        c = n-a-b
	        #large = 0
	        if a*a+b*b == c*c and a+b+c == n:
	            if a*b*c > large:
	                large = a*b*c
	    if large != 0:
	        print(large)  
	    else:
	        print(-1) 